service/dap/server_test.go

```diff
@@ -2037,6 +2037,73 @@ func TestSetBreakpoint(t *testing.T) {
 	})
 }
 
+func checkHitBreakpointIds(t *testing.T, se *dap.StoppedEvent, reason string, id int) {
+	if se.Body.ThreadId != 1 || se.Body.Reason != reason || len(se.Body.HitBreakpointIds) != 1 || se.Body.HitBreakpointIds[0] != id {
+		t.Errorf("got %#v, want Reason=%q, ThreadId=1, HitBreakpointIds=[]int{%d}", se, reason, id)
+	}
+}
+
+// TestHitBreakpointIds executes to a breakpoint and tests that
+// the breakpoint ids in the stopped event are correct.
+func TestHitBreakpointIds(t *testing.T) {
```

Comment:
This feature should apply to all breakpoints, right? So why not check an additional setting no the breakpoints int the other breakpoint-dedicated tests we already have?

Reply:
This should apply to all breakpoints, but I thought checking that each ID lined up would be too much of a hassle to do in every test, since we have to then indicated which breakpoint we wanted to hit for every ID. 

I guess this could be done with a search in the SetBreakpointsResponse, but I thought it would add unnecessary complexity to the other tests.

Reply:
I did not mean doing this in all test functions that use breakpoints to test other things. I meant adding a couple of extra lines to the test functions dedicated to breakpoints: TestPreSetBreakpoint, TestSetBreakpoint, TestSetFunctionBreakpoint.